tests/test_client.py

```diff
@@ -64,94 +64,84 @@ def test_index_create_with_multiple_hashes(client, user):
             'sha1': 'fdbbca63fbec1c2b0d4eb2494ce91520ec9f55f5'
         }
     }
-    assert client.post(
-        '/index/',
-        data=json.dumps(data),
-        headers=user).status_code == 200
 
-def test_index_create_with_valid_did(client, user):
+    result = swg_client.add_entry(data)
+    assert result.did
+
+
+def test_index_create_with_valid_did(swg_client):
     data = {
         'did':'3d313755-cbb4-4b08-899d-7bbac1f6e67d',
         'form': 'object',
         'size': 123,
         'urls': ['s3://endpointurl/bucket/key'],
         'hashes': {'md5': '8b9942cf415384b27cadf1f4d2d682e5'}}
 
-    r = client.post(
-        '/index/',
-        data=json.dumps(data),
-        headers=user)
+    result = swg_client.add_entry(data)
+    assert result.did == '3d313755-cbb4-4b08-899d-7bbac1f6e67d'
 
-    assert r.status_code == 200
-    assert r.json['did'] == '3d313755-cbb4-4b08-899d-7bbac1f6e67d'
 
-def test_index_create_with_invalid_did(client, user):
+def test_index_create_with_invalid_did(swg_client):
     data = {
-        'did':'3d313755-cbb4-4b0fdfdfd8-899d-7bbac1f6e67dfdd',
+        'did': '3d313755-cbb4-4b0fdfdfd8-899d-7bbac1f6e67dfdd',
         'form': 'object',
         'size': 123,
         'urls': ['s3://endpointurl/bucket/key'],
         'hashes': {'md5': '8b9942cf415384b27cadf1f4d2d682e5'}}
 
-    assert client.post(
-        '/index/',
-        data=json.dumps(data),
-        headers=user).status_code == 400
+    try:
```

Comment:
I like the with.exception pattern that pytest gives better, but it doesn't matter.